source/components/recipe-search/recipe-search.js

```diff
@@ -231,25 +231,209 @@ class RecipeSearch extends YummyRecipesComponent {
+
+          // Get page number before the clicked on page number
+          let lastPageNum = "";
+
+          let pageChunk = url.slice(url.indexOf("?page="), url.length);
+          let equalIndex = pageChunk.indexOf("=");
+
+          for (let i = equalIndex + 1; i < pageChunk.length; i++) {
+            lastPageNum = lastPageNum + pageChunk[i];
+          }
+
+          // Remove the active class from the prev page number
+          thisContainer.shadowRoot
+            .getElementById(lastPageNum)
+            .classList.remove("active");
+
+          anchorTag.classList.add("active");
+
+          // Gets the URL without ?page=, then adds it back with the current page number
+          const newUrl =
+            url.slice(0, url.indexOf("?page=")) + "?page=" + currPage;
+
+          history.pushState({}, "", newUrl);
+        });
+
+        // Append the tag to paginationDiv
+        pagDiv.append(anchorTag);
+      }
+
+      // Create right arrow icon for navigation
+      let rightArrow = document.createElement("a");
+      rightArrow.innerHTML = "&#8594;";
+      rightArrow.classList.add("pagination-arrow");
+
+      // If clicked on, click next page anchor tag
+      rightArrow.addEventListener("click", () => {
+        let url = window.location.href;
+
+        let currPageNum = "";
+
+        let pageChunk = url.slice(url.indexOf("?page="), url.length);
+        let equalIndex = pageChunk.indexOf("=");
+
+        for (let i = equalIndex + 1; i < pageChunk.length; i++) {
+          currPageNum = currPageNum + pageChunk[i];
+        }
+
+        currPageNum = parseInt(currPageNum);
+
+        // The page after the last page DNE
+        if (currPageNum == pageCount) {
+          return;
+        } else {
+          let nextPage = currPageNum + 1;
+          thisContainer.shadowRoot
+            .getElementById(currPageNum)
+            .classList.remove("active");
+
+          thisContainer.shadowRoot
+            .getElementById(nextPage)
+            .classList.add("active");
+
+          thisContainer.shadowRoot.getElementById(nextPage).click();
+        }
+      });
+      pagDiv.append(rightArrow);
+
+      // Populate page initially with first recipePerPage recipes
+      for (let i = 0; i < recipePerPage; i++) {
+        /**
+         * Closes filters form when clicking on "X" icon.
+         */
+        this.shadowRoot
+          .getElementById("recipe-card-grid")
+          .appendChild(
+            this.createRecipeCard(searchRecipe[i].recipe, searchRecipe[i].index)
+          );
+      }
+
+      // Set url to have page = 1 on initial load
+      const url = window.location.href;
+
+      // Gets the URL without ?page=, then adds it back with the current page number
+      const newUrl = url + "?page=" + 1;
+
+      history.pushState({}, "", newUrl);
+
+      this.shadowRoot.getElementById("1").classList.add("active");
+    }
+    // For the case that pagination is not required
+    else {
```

Comment:
Please rearrange as below:
```suggestion
    } else {
    // For the case that pagination is not required
```